### PR TITLE
fix: improve performance combining coverage on large codebases

### PR DESCRIPTION
**Pull request by koenvangeert**
Remove the constant reading and writing of the coverage object to disk for each test.

Keep it in memory and then write it once after each suite to get the correct report out.
We can optimise further by only writing to disk after all suites are done but the performance gains for this are a lot less.

Some measurements from the codebase I'm testing this on.

Before the change:
combineCoverage: 855.495ms

After:
combineCoverage: 295.321ms

### CODE REVIEW

**Comment by jennifer-shehane** [comment]
@koenvangeert Could you sign our CLA?

**Comment by koenvangeert** [comment]
@jennifer-shehane Can I do anything else to help this move forward?

File: task.js [hunk 1]
```diff
@@ -148,21 +155,9 @@ const tasks = {
...
-      ? JSON.parse(readFileSync(nycFilename, 'utf8'))
-      : {}
-
-    // previous code coverage object might have placeholder entries
-    // for files that we have not seen yet,
-    // but the user expects to include in the coverage report
-    // the merge function messes up, so we should remove any placeholder entries
-    // and re-insert them again when creating the report
-    removePlaceholders(previousCoverage)
+    console.log(coverageMap.toJSON())
```
Comment (emilyrohrbough) [inline task.js:158]:
```suggestion
```

Comment (emilyrohrbough) [reply task.js:158]: Why was the removePlaceholders logic removed?

Comment (jennifer-shehane) [reply task.js:158]: @koenvangeert Sorry for the long response time. Are you able to respond to Emily's review?

Comment (koenvangeert) [reply task.js]: @jennifer-shehane Sorry for my long response time 🙇 . I fixed the comments, can I do something do push this along?

Comment (jennifer-shehane) [reply task.js]: We'll take another look. Thanks!

**Pull request merged**